deploy/stacks/cognito.py

```diff
@@ -50,6 +53,41 @@ def __init__(
         cfn_user_pool.user_pool_add_ons = cognito.CfnUserPool.UserPoolAddOnsProperty(
             advanced_security_mode='ENFORCED'
         )
+
+        # Create IP set if IP filtering enabled in CDK.json
+        ip_set_regional = None
+        if custom_waf_rules and custom_waf_rules.get('allowed_ip_list'):
+            ip_set_regional = wafv2.CfnIPSet(
+                self,
+                'DataallRegionalIPSet-Cognito',
+                name=f'{resource_prefix}-{envname}-ipset-regional',
```

Comment:
The name of the IP set is the same one as for the WAF of API GW in `lambda_api.py`. This will cause `WAFDuplicateItemException: AWS WAF couldn’t perform the operation because some resource in your request is a duplicate of an existing one.` when `allowed_ip_list` is defined

Reply:
We can create 2 different IP sets (as we are trying to do now) or we can think of how to merge it in one. I personally see the value fo having 2 separate IP sets that allows this feature to be extended and restrict IP addresses differently for Cognito and for API GW

Reply:
good catch - updating the PR now